### DISCUSSION TITLE
How to add examples for parameters for swagger docs, from the endpoint definition in .py?

### DISCUSSION THREAD

**Discussion by joshjm** [fastapi/fastapi#6885] [Questions] [answered]
### First check

* [x] I added a very descriptive title to this issue.
* [x] I used the GitHub search to find a similar issue and didn't find it.
* [x] I searched the FastAPI documentation, with the integrated search.
* [x] I already searched in Google "How to X in FastAPI" and didn't find any information.
* [x] I already read and followed all the tutorial in the docs and didn't find an answer.
* [x] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
* [x] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
* [x] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).
* [x] After submitting this, I commit to one of:
    * Read open issues with questions until I find 2 issues where I can help someone and add a comment to help there.
    * I already hit the "watch" button in this repository to receive notifications and I commit to help at least 2 people that ask questions in the future.
    * Implement a Pull Request for a confirmed bug.

### Example

Here's a self-contained, example with my use case. :

```Python
from fastapi import FastAPI

app = FastAPI()

@app.post("/{name}")
def read_root(name):
    """
    Some descriptive doc-string
    """
    return f"Hello {name}"
```

### Description

* Open the browser and call the endpoint `/`.
* Navigate to the /docs
* note that swagger can allow you to set examples for the query param, but there is no obvious way to do this with swagger docs generated from fast api. 
### Environment

osx, plus from pip freeze:
```
certifi==2020.12.5
click==7.1.2
fastapi==0.63.0
h11==0.12.0
pydantic==1.8
starlette==0.13.6
typing-extensions==3.7.4.3
uvicorn==0.13.4
```

* Python version: 3.8

### Additional context

Is there some pattern with pydoc or something with setting examples that I am mising? or with pydantic?

I would like to be able to use the swagger docs for some endpoint development, so it would be great to be able to use swagger to remember an example UUID for example, so speed up the tests, and let me skip duplicating all the endpoints to postman.

**Comment by Kludex** [comment, marked as answer]
I don't remember if this is written somewhere... I don't think it is.

```python
from fastapi import FastAPI, Path

app = FastAPI()

@app.post("/{name}")
def read_root(name: str = Path(..., example="I can't believe is that easy!")):
    """
    Some descriptive doc-string
    """
    return f"Hello {name}"
```

![image](https://user-images.githubusercontent.com/7353520/109828662-22956f00-7c3d-11eb-9fd5-b20c4aa98e14.png)

**Comment by joshjm** [comment]
oh man, thats awesome. just what i was looking for. 
Now im feeling greedy, and looking at fastapi>openapi>models.py it seems that there may also be a way to add multiple examples
```
class ParameterBase(BaseModel):
    description: Optional[str] = None
    required: Optional[bool] = None
    deprecated: Optional[bool] = None
    # Serialization rules for simple scenarios
    style: Optional[str] = None
    explode: Optional[bool] = None
    allowReserved: Optional[bool] = None
    schema_: Optional[Union[Schema, Reference]] = Field(None, alias="schema")
    example: Optional[Any] = None
    examples: Optional[Dict[str, Union[Example, Reference]]] = None
    # Serialization rules for more complex scenarios
    content: Optional[Dict[str, MediaType]] = None
```

**Comment by Kludex** [comment]
I'm too tired, so it may be non sense. But maybe you're waiting for this? https://github.com/tiangolo/fastapi/pull/1267

**Comment by joshjm** [comment]
ahh yep; a one two punch of couldnt find docs on making a single example, + then wanting what the PR provides. I'll close and ~~mark as dupe~~(would if i could). thanks for your help! i'll try and make some comments on that pr

**Comment by tiangolo** [comment]
Thanks for the help here @Kludex ! 👏  🙇 

Thanks for reporting back and closing the issue @joshjm  👍

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
